src/input/pointer/mod.rs

```diff
@@ -435,6 +435,23 @@ impl<D: SeatHandler + 'static> PointerHandle<D> {
         self.inner.lock().unwrap().location
     }
 
+    /// Update the current location of this pointer in the global space,
+    /// without sending any event and without updating the focus.
+    ///
+    /// This is useful when the pointer is only moved by relative events,
+    /// such as when a pointer lock is held by the focused surface.
+    /// The client can give us a cursor position hint, which corresponds to
+    /// the actual location the client may be rendering a pointer at.
+    /// This position hint should be used as the initial location
+    /// when the pointer lock is deactivated.
+    ///
+    /// The next time [Self::motion] is called, the focus will be
+    /// updated accordingly as if this function was never called.
+    /// Clients will never be notified of a location hint.
+    pub fn set_location_hint(&self, location: Point<f64, Logical>) {
+        self.inner.lock().unwrap().location = location;
+    }
```

Comment:
The function's doc comment is really clear :ok_hand: , but I'm not a fan of the name.
It always overrides the pointer location, so calling it a hint seems misleading.

I don't have any good ideas tho, perhaps `set_location_raw`?

Reply:
I do agree that the name could be... different... but also, it only overrides the pointer location in the eyes of the compositor. It's a "hint" for the next time the compositor wants to use the absolute location (typically, this is called while a `LockedPointer` is held and we're only sending relative events) in the sense that it only modifies the compositor's view of `PointerHandle::current_location()`.

It doesn't actually update the pointer location in the Wayland protocol; changing this field does nothing to notify any client about anything, nor does any focus state change. While it does always overwrite a field called `location`, i don't think it always overrides "The Location"

Reply:
What I mean is that for example in Anvil calling this will cause the cursor to be rendered in a different place on the screen. So yes it does not update the location in Wayland, but it does everywhere else (if compositor uses this location value, like Anvil or Cosmic)

Reply:
This is the intended behaviour of the "cursor position hint" though, is it not? It updates where the cursor should visually be.

Most clients use of this will just be to move the cursor to a different place on its own surface, when they have the cursor surface hidden. Then nothing happens visually, unless the pointer lock is deactivated. If a client sets a position hint that's out of bounds for its surface, i don't see the problem with a compositor rendering the cursor at the hinted location (and perhaps indirectly causing the surface with the lock to lose focus). This is compositor-defined how they want to deal with that case.

It's also relevant to note that this function is called by the compositor, not by the protocol handler. If its semantics are undesirable, the compositor can choose to call it at a later point or just never call it, dealing with pointer hints however they want to deal with it.

Reply:
Think about this from a perspective of a compositor rather than from a protocol perspective (in fact, the input module is not Wayland specific, it's just an input event propagation abstraction).

The purpose of this function is to set The location without notifying input listeners about that change.

Therefore, this function is not setting a hint. It is setting the actual location in response to a hint. (which is fine btw)

If a function is modifying the single source of truth for pointer location, it can't be called `set_hit`.

I would be fine with `fn set_location`, `fn set_location_silent`, `fn set_location_no_notify`, or `fn set_location_raw`

Reply:
alright, i've renamed the function in https://github.com/Smithay/smithay/pull/1551/commits/f4c0842bbf3de89e7eb2ca60af4e17ee59d8a348